src/Data/Text/Utf8.hs

```diff
@@ -199,6 +148,17 @@ unicode2utf8 c
     | c < 0x10000 = [0xe0 .|. (c `shiftR` 12), 0x80 .|. (0x3f .&. (c `shiftR` 6)), 0x80 .|. (0x3f .&. c)]
     | otherwise   = [0xf0 .|. (c `shiftR` 18), 0x80 .|. (0x3f .&. (c `shiftR` 12)), 0x80 .|. (0x3f .&. (c `shiftR` 6)), 0x80 .|. (0x3f .&. c)]
```

Comment:
It's likely faster to use [`utf8Length`](https://hackage.haskell.org/package/text-2.0/docs/Data-Text-Internal-Encoding-Utf8.html#v:utf8Length) and then switch between `ord2`, `ord3`, `ord4`.

Reply:
`unicode2utf8` and `decodeUtf8` are used in our test suite only, they are optimized for legibility rather than speed. This probably warrants a comment or we could move them to the `Orphans` module in `./test`.